harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -1059,12 +1054,17 @@ def _validate(self, searcher_op: Optional[core.SearcherOperation] = None) -> Dic
             # common than evaluate_batch() and we can't know how the user processed their
             # validation data.
             if self._evaluate_batch_defined():
+                # Reshape and sum.
+                # TODO: remove the type directive once we upgrade to mypy >= 1.7.0
+                inputs_total, batches_total = [sum(n) for n in zip(*input_counts)]  # type: ignore
```

Comment:
input_counts doesn't seem to be defined in the evaluate_full_dataset codepath

Reply:
The code is kind of unfortunately structured and pretty hard to figure out:
```
1. if evaluate_batch:
2.   do stuff in a managed way
3.   if chief:
4.     do managed stuff relevant to the chief
5. else:  # evaluate_full
6.   do whatever stuff evaluate_full says to do
7. 
8. if chief:
9.   if evaluate_batch:
10.     report batch-specific detail
11.  report general stuff
```    
Before patch:
* `num_inputs` defined / calculated line 2
* `input_counts` defined line 2 (from gathered `num_inputs`)
* `num_inputs` re-defined line 4 when evaluate_batch (from `input_counts`)
* `num_inputs` re-defined line 6 when evaluate_full (from `len(validation_loader)`)
* `num_inputs` (per its second definition) used line 10 and nowhere else

Pre-patch, `num_inputs` was being defined when both evaluate_batch and evaluate_full, but only used during evaluate_batch.

After patch:
* `num_inputs` defined / calculated line 2
* `input_counts` defined line 2 (from gathered `num_inputs`)
* `inputs_total` defined line 10 (from `input_counts`)
* `inputs_total` used line 10

Effects of this patch:
* `num_inputs` not defined during evaluate_full (which is fine, because it hadn't been used there)
* gathered calculation from `input_counts` moved to where it's used (line 10)
* gathered calculation from `input_counts` given a new name so `num_inputs` isn't overloaded